### ISSUE TITLE
systemctl enable/disable loop in centos 7 using version 2.6.8

### ISSUE THREAD

**Issue by jslusher** [chef-cookbooks/docker#701]
### Cookbook version

2.6.8
### Chef-client version

12.4.1
### Platform Details

centos 7
EC2 
### Scenario:

Initial converge
### Steps to Reproduce:

Using centos 7 with version 2.6.8 of the `docker_service` lwrp:

```
docker_service 'default' do
  insecure_registry 'registry.prod.example.com'
  exec_opts ['native.cgroupdriver=cgroupfs']
  if direct_lvm['enabled'] then
    storage_driver 'devicemapper'
    storage_opts [
      'dm.datadev=/dev/vg-docker/data',
      'dm.metadatadev=/dev/vg-docker/data2'
    ]
  end
  action [:create, :start]
end
```
### Expected Result:

Converge enters into a loop in systemctl that disables, daemonctl-reload and enables the docker service.
### Actual Result:

https://gist.github.com/jslusher/4d643b2969b9a28c4144a493caa13dd1

**Comment by Fryie** [comment]
Happens on Debian 8 too.

**Comment by Chili-Man** [comment]
I'm running into this issue as well with version `2.9.3` of the cookbook: 
https://gist.github.com/Chili-Man/68fec37d8628fd0648fcd6b80a0d9e5b

**Comment by shatil** [comment]
I'm surprised `systemctl` worked at all. I tried on CentOS and on Debian, but `systemd` related things just do not work for me, I'm guessing because I haven't done my homework (https://lwn.net/Articles/676831/).

**Comment by someara** [comment]
I've reverted 4d6956850f79dfe62d25bbfa9bda126f4e80ea1c.

**Comment by someara** [comment]
Released as 2.11.1
